File: tools/create-order.py
import requests
import json

url = "http://localhost:3000/api/order/create"

def create_order(order):
    payload = json.dumps(order)
    headers = {
        'Content-Type': 'application/json'
    }
    response = requests.request("POST", url, headers=headers, data=payload)
    print(response.text.encode('utf8'))


if __name__ == '__main__':
    for id in range(1, 20):
        order = {
            "id": str(id),
            "name": "防护服" + str(id),
            "client": "武汉市红十字会",
            "receiver": "解贝",
            "source": "武汉会展中心",
            "destination": "珞瑜路152号 华中师范大学",
            "description": "一次性防护服 官员专用"
        }
        create_order(order)
        

File: shipper-client/createOrder.js
/*
 * SPDX-License-Identifier: Apache-2.0
 */

'use strict';

const { Gateway, Wallets } = require('fabric-network');
const fs = require('fs');
const path = require('path');

async function main() {
    try {
        // load the network configuration
        const ccpPath = path.resolve(__dirname, 'connection.json');
        let ccp = JSON.parse(fs.readFileSync(ccpPath, 'utf8'));

        // Create a new file system based wallet for managing identities.
        const walletPath = path.join(process.cwd(), 'wallet');
        const wallet = await Wallets.newFileSystemWallet(walletPath);
        console.log(`Wallet path: ${walletPath}`);

        // Check to see if we've already enrolled the user.
        const identity = await wallet.get('admin');
        if (!identity) {
            console.log('An identity for the user "appUser" does not exist in the wallet');
            console.log('Run the registerUser.js application before retrying');
            return;
        }

        // Create a new gateway for connecting to our peer node.
        const gateway = new Gateway();
        await gateway.connect(ccp, { wallet, identity: 'admin', discovery: { enabled: true, asLocalhost: true } });

        // Get the network (channel) our contract is deployed to.
        const network = await gateway.getNetwork('logistic-channel');

        // Get the contract from the network.
        const contract = network.getContract('logistic');

        request = {
            id: '1',
            name: '货物1',
            source: '武汉市华中师范大学',
            destination: '武汉市华中科技大学',
            client: '发货人一号',
            receiver: '收货人二号',
            description: '绝密绝密绝密！'
        }

        const jsonStr = JSON.stringify(request)
        // Submit the specified transaction.
        await contract.submitTransaction('createOrder', jsonStr);
        console.log(`Transaction has been submitted. request: ${jsonStr}`);

        await gateway.disconnect();

    } catch (error) {
        console.error(`Failed to submit transaction: ${error}`);
        process.exit(1);
    }
}

main();
